engine/src/settings/settings.cpp

```diff
@@ -96,6 +103,54 @@ void Settings::merge(const Settings& settingsToMerge)
     }
 }
 
+static inline void saveSettingsAsJson(const Settings& settings, Stream& stream)
+{
+    stream.put('{');
+    bool first = true;
+    for (const auto& value : settings.getValues())
+    {
+        if (!first)
+        {
+            stream.put(',');
+        }
+        first = false;
+
+        char* pEnd;
+        strtod(value.second.c_str(), &pEnd);
+
+        // Use "" if second value is not number nor boolean
+        if (pEnd != value.second.c_str() + value.second.length() && (value.second != "true" && value.second != "false"))
+        {
+            stream.printf("\n\t\"{}\": \"{}\"", value.first.c_str(), value.second.c_str());
+        }
+        else
+        {
+            stream.printf("\n\t\"{}\": {}", value.first.c_str(), value.second.c_str());
+        }
+    }
+    stream.print("\n}\n");
+}
+
+bool Settings::save()
+{
```

Comment:
```suggestion
bool Settings::save(std::string_view path)
{
```
I would change this to receive the virtual file system path where to save to, and by default set it to `/settings.json`.
This gives the caller more flexibility. Internally then, you wouldn't mount an archive, you would only keep the last three lines. Since the settings file is already mounted at startup anyway the logic in the beginning is unnecessary